src/msg/async/ProtocolV2.cc

```diff
@@ -2671,6 +2675,17 @@ CtPtr ProtocolV2::handle_existing_connection(const AsyncConnectionRef& existing)
     return WRITE(wait, "wait", read_frame);
   }
 
+  if (peer_global_seq == 0 && 
+      exproto->client_cookie && client_cookie && 
+      exproto->client_cookie != client_cookie) {
+    ldout(cct, 1) << __func__ << " client has clearly restarted (global_seq=0, cookie changed), "
```

Comment:
Yeah, for the client-reset case we aren't choosing the reconnect path as the `server_cookie` should be zero.

```cpp
CtPtr ProtocolV2::start_session_connect() {
  if (!server_cookie) {
    ceph_assert(connect_seq == 0);
    state = SESSION_CONNECTING;
    return send_client_ident();               
  } else {  // reconnecting to previous session
    state = SESSION_RECONNECTING;
    ceph_assert(connect_seq > 0);
    return send_reconnect();
  }
}
```

A client restart (in contrast to a connection reset alone) thrashes the entire state-at-client. I don't see business in trying to ressurrect the `exproto` in such conditions. @idryomov, what do you think?